src/google/protobuf/stubs/io_win32.cc

```diff
@@ -155,12 +155,13 @@ string normalize(string path) {
 
   static const string dot(".");
   static const string dotdot("..");
+  const string& cpath = path;
```

Comment:
how about using const char* p looping on path.c_str() instead? This avoids mixed use of cpath and path in the parsing code.

Reply:
Yeah, that's probably preferable. Also may have better performance than calling `operator[]` every time